Sources/MongoSwift/BSON/Document.swift

```diff
@@ -420,7 +422,7 @@ extension Document: BSONValue {
         bson_iter_document(&iter.iter, &length, document)
 
         guard let docData = bson_new_from_data(document.pointee, Int(length)) else {
-            throw MongoError.bsonDecodeError(message: "Failed to create a bson_t from document data")
+            throw RuntimeError.internalError(message: "Failed to create a bson_t from document data")
```

Comment:
since you are updating this line anyway, could we give this a better error message? it exposes a libbson internal users shouldn't need to think about. 
not sure how much we want to rely on libbson internals since this is documented as just returning null on an unnamed failure. but at the moment anyway it looks like this just returns null when there are issues with the length (see [here]( https://github.com/mongodb/mongo-c-driver/blob/16be1bf48fe3378a593adcf84fe6bb1c8c8cd3f5/src/libbson/src/bson/bson.c#L2095))